### ISSUE TITLE
Readonly Macaroon can't listen to RPC Middleware

### ISSUE THREAD

**Issue by gkrizek** [lightningnetwork/lnd#6839]
### Background

When trying to register with the RPC Middleware using the readonly.macaroon, I get a `permission denied` error.

### Your environment

* LND 0.15.1.rc1
* MacOS and Ubuntu
* btcd 0.23.1

### Steps to reproduce

Attempt to register with the RPC Middleware with a readonly.macaroon.

### Expected behaviour

I think the readonly macaroon should be able to listen to the RPC Middleware. Obviously it should not be able to make and changes or adjustments in the middleware. But simple being able to _read_ the stream seems like it's in line with what a readonly macaroon can do. I want to be able to see the RPC requests in my node without letting the process make modifications.

### Actual behaviour

The only way I can see the RPC requests is via an Admin macaroon.

**Comment by ellemouton** [comment]
yeah so currently, access to the  `RegisterRPCMiddleware` URI also gives you the ability to create a full (non-read-only) interceptor which does require an admin macaroon since with the full interceptor you have the ability to re-write any request & response. The `read-only` option is just a field in the request for that method. 

Agreed that it would be cool to be able to have the read only option with a read-only mac. This would require a new `RegisterReadOnlyRPCMiddleware` method.

**Comment by ellemouton** [comment]
If you agree, then I think this issue should be renamed to "feature request: add RegisterReadOnlyRPCMiddleware method to rpc"

**Comment by gkrizek** [comment]
@ellemouton Thanks for the response. This makes sense. I can have someone on our team look into implementing this. Before I rename this or start work I want to confirm that creating an entire new method is the best solution instead of modifying the existing method? I really haven't dug into the code here at all, just wanting to confirm that's the preferred pattern. Thanks!

**Comment by ellemouton** [comment]
cool, yeah let's get a second opinion from @guggero. But as far as I know, a macaroon can either provide access or not provide access to a specific method but cant enforce how that method is used. 

Btw, are you using this endpoint via LND's rpc server directly or are you using it via `lndclient`?

**Comment by guggero** [comment]
I agree that we probably shouldn't start adding business logic to our macaroon interceptor to accept different macaroons on the same RPC depending on their usage.
So a `RegisterReadOnlyRPCMiddleware` RPC that only accepts read-only registrations is probably the simplest option. But it won't make the usage/documentation very clear if we still allow the normal middleware RPC to allow read-only registrations. But to not make breaking changes to the API that's likely the only way.

**Comment by gkrizek** [comment]
@ellemouton We are using it via `lndclient`

@guggero Ok this makes sense. A new RPC method it is.

I'll have someone from our team take a look to try to understand the level of effort here. Thanks!
